docs/ReferenceGuides/UnderscoredAttributes.md

```diff
@@ -0,0 +1,428 @@
+expression type checker is considering overloads, it will prefer a solution
+with fewer `@_disfavoredOverload` declarations over one with more of them.
+
+Historically, it was designed to help with `ExpressibleByXYZLiteral` types.
+The type checker strongly prefers to give literals their default type
+(e.g. `Int` for `ExpressibleByIntegerLiteral`,
+`String` for `ExpressibleByStringLiteral`, etc.).
+If an API should prefer some other type, but accept the default too,
+marking the declaration taking the default type with `@_disfavoredOverload`
+gives the desired behavior:
+
+```swift
+extension LocalizedStringKey: ExpressibleByStringLiteral { ... }
+
+extension Text {
+  // We want `Text("foo")` to use this initializer:
+  init(_ key: LocalizedStringKey) { ... }
+
+  // But without @_disfavoredOverload, it would use this one instead,
+  // because that lets it give the literal its default type:
+  @_disfavoredOverload init<S: StringProtocol>(_ str: S) { ... }
+}
+```
+
+`@_disfavoredOverload` is a stop-gap design; eventually the type-checker
+behavior should be fixed and the attribute should be removed/made into a no-op.
+
+Its main use case is to work around clearly undesirable overload resolution,
+not to arbitrarily manipulate overload priority.
+
+## `@_dynamicReplacement(for:)`
+
+Marks a function as the dynamic replacement for another `dynamic` function.
+This is similar to method swizzling in other languages such as Objective-C,
+except that the replacement happens at program start (or loading a shared
+library), instead of at an arbitrary point in time.
+
+For more details, see
+[the forum post on Dynamic method replacement](https://forums.swift.org/t/dynamic-method-replacement/16619).
+
+## `@_distributedActorIndependent`
+
+## `@_effects(effectname)`
+
+Tells the compiler that the implementation of the defined function is limited
+to certain side effects. The attribute argument specifies the kind of side
+effect limitations that apply to the function including any other functions
+it calls. This is used to provide information to the optimizer that it can't
+already infer from static analysis.
+
+Changing the implementation in a way that violates the optimizer's assumptions
+about the effects results in undefined behavior.
+
+For more details, see [OptimizerEffects.rst](/docs/Proposals/OptimizerEffects.rst).
+
+## `@_exported`
+
+Re-exports all declarations from an imported module.
+
+This attribute is most commonly used by overlays.
+
+```swift
+// module M
+func f() {}
+
+// module N
+@_exported import M
+
+// module P
+import N
+func g() {
+  N.f() // OK
+}
+```
+
+## `@_fixed_layout`
+
+## `@_hasInitialValue`
+
+## `@_hasMissingDesignatedInitializers`
+
+Indicates that there may be designated initializers that are
+not printed in the swiftinterface file for a particular class.
+
+This attribute is needed for the initializer model to maintain correctness when
+library evolution is enabled. This is because a class may have non-public
```

Comment:
Could you describe why you think having a link specifically for library evolution would be valuable here? (There is one more place below mentioning library evolution under _spi, do you think I should link it there too?)

Generally, I'm in favor of having more links rather than fewer, but in this case, this document is not trying to explain much because it assumes that you mostly know what you're doing. E.g. it doesn't provide links for swiftinterfaces.

Also "Swift Library Evolution" gives pretty good search hits if you google it.

Reply:
We have a document about library evolution, if I recall, as a blog post; if a user is unfamiliar with the term it sounds similar to "Swift evolution" and can be be hard to google, in my view, and there's pretty low effort required to provide a link to the relevant document.

Reply:
I think our audience should strive to accommodate whoever might need to reference the meaning of the attribute after stumbling across it. So, `@_marker` is likely to have a broader audience than `@_staticInitializeObjCMetadata`.

Reply:
Added link.